services/dedup/dedup.go

```diff
@@ -169,27 +169,27 @@ func (d *DedupHandleT) gcBadgerDB() {
 	}
 }
 
-func (d *DedupHandleT) writeToBadger(messageIDs []string) {
-	err := d.badgerDB.Update(func(txn *badger.Txn) error {
-		for _, messageID := range messageIDs {
-			e := badger.NewEntry([]byte(messageID), nil).WithTTL(*d.window)
-			if err := txn.SetEntry(e); err == badger.ErrTxnTooBig {
-				_ = txn.Commit()
-				txn = d.badgerDB.NewTransaction(true)
-				_ = txn.SetEntry(e)
+func (d *DedupHandleT) writeToBadger(messageIDs []string) error {
+	txn := d.badgerDB.NewTransaction(true)
+	for _, messageID := range messageIDs {
+		e := badger.NewEntry([]byte(messageID), nil).WithTTL(*d.window)
+		if err := txn.SetEntry(e); err == badger.ErrTxnTooBig {
```

Comment:
We should also check for err != nil. 

Unfortunately badger db examples don't contain proper error handling and this one is easy to miss

Reply:
yes indeed, missed that